### DISCUSSION TITLE
Android Debug Mode

### DISCUSSION THREAD

**Discussion by marchbold** [airsdk/Adobe-Runtime-Support#1923] [Q&A] [answered]
I've been trying to debug an Android application on an Amazon device, however I'm having some issues with the Amazon SDK believing the build with target `apk-debug` is a release version. 

However if I open up the AndroidStudioProject and debug from there the SDK picks it up correctly as a debug build.

I'm wondering what could be happening here or what is the difference between the `apk-debug` and a debug from Android studio? 

> Note: this could just be an issue with the Amazon SDK, but just thought I'd ask.

**Comment by ajwfrost** [comment, marked as answer]
Hi

Curiously, it looks like we're always calling `assembleRelease` (or `bundleRelease`) as the Gradle command even if using `apk-debug` (or `aab-debug`). The differences are in our internal runtime files where we enable some other changes to include debug capabilities for the runtime.

It would seem to be better for us to use the `apk` target type with `assembleRelease`, and `apk-debug` with `assembleDebug` as well (and likewise for AAB..)

Will get that very quick change done for our next release...!

thanks

**Reply by marchbold** [reply]
Ah brilliant, I had a feeling it might have been something simple like that! Thanks for making this change.

**Reply by marchbold** [reply]
For anyone coming across this, the issue was resolved in 33.1.1.889.
